TrafficCapture/buildSrc/src/main/groovy/org/opensearch/migrations/common/CommonUtils.groovy

```diff
@@ -0,0 +1,91 @@
+package org.opensearch.migrations.common
+
+import org.gradle.api.tasks.Copy
+import org.gradle.api.Project
+import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+
+class CommonUtils {
+    static def calculateDockerHash(def projectName, def project) {
+        def MessageDigest = java.security.MessageDigest
+        def digest = MessageDigest.getInstance('SHA-256')
+        project.fileTree("src/main/docker/${projectName}")
+                .each { file ->
+                    file.withInputStream { is ->
+                        byte[] buffer = new byte[1024]
+                        int read
+                        while ((read = is.read(buffer)) != -1) {
+                            digest.update(buffer, 0, read)
+                        }
+                    }
+                }
+        return digest.digest().encodeHex().toString()
+    }
+
+    static def copyArtifact(Project project, String projectName) {
+        def dockerBuildDir = "build/docker/${projectName}"
+        def artifactsDir = "${dockerBuildDir}/jars"
+        project.task("copyArtifact_${projectName}", type: Copy) {
+            dependsOn ":${projectName}:build"
+            dependsOn ":${projectName}:jar"
+            if (projectName == "trafficCaptureProxyServerTest") {
+                include "*.properties"
+                def jmeterPropertiesFile = project.file("src/main/docker/${projectName}/jmeter.properties")
+                if (jmeterPropertiesFile.exists()) {
+                    from jmeterPropertiesFile.absolutePath
+                    into "${dockerBuildDir}"
+                }
+            }
+            from { project.project(":${projectName}").configurations.findByName("runtimeClasspath").files }
+            from { project.project(":${projectName}").tasks.getByName('jar') }
+            into artifactsDir
+            include "*.jar"
+            duplicatesStrategy = 'WARN'
+        }
+    }
+
+    static def createDockerfile(Project project, String projectName, Map<String, String> baseImageProjectOverrides, Map<String, String> dockerFilesForExternalServices) {
```

Comment:
This will need some eventual refactoring - but not for this task.  There are a lot of assumptions that we're building precisely 2 kinds of dockerfiles.  We also could be better off w/ static Dockerfiles for each of these.  Again, those changes would not be for this PR.